### PR TITLE
build: ensure index.json is actually valid JSON before uploading

### PR DESCRIPTION
**Pull request by MarshallOfSound**
I managed to recover the damage from yesterday but this will ensure that we never push bad JSON again 👍 

Notes: no-notes

### CODE REVIEW

File: script/upload-index-json.py [hunk 1]
```diff
@@ -15,16 +16,28 @@
...
 
-    return urllib2.urlopen(
+    proposed_content = urllib2.urlopen(
       request
     ).read()
+
+    if is_json(proposed_content):
+      return proposed_content
+    print("bad attempt")
+    raise Exception("Failed to fetch valid JSON from the metadumper service")
```
Comment (ckerr) [inline script/upload-index-json.py]: Would it be useful to append the json.loads error message to this for postmortems?

Comment (MarshallOfSound) [reply script/upload-index-json.py]: @ckerr I'm uncertain as to what that error message may contain (E.g. Secret keys from the metadumper) The error would be logged on the metadumper anyway so we should be 🆗 here without logging

**Pull request merged**